### PR TITLE
[QT-517] Skip builds for docs PRs

### PR DESCRIPTION
**Pull request by jaymalasinha**
Skip builds and Enos tests for docs PRs.  Following https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks we conditionally skip the jobs.

### CODE REVIEW

**Comment by jaymalasinha** [comment]
example of skipped build/test workflow https://github.com/hashicorp/vault/actions/runs/4746096146

File: .github/scripts/verify_changes.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+event_type=$1
+base_ref=$2
+
+changed_dir=""
+
+git fetch --no-tags $base_ref
+
+if [[ "$1" == "pull_request" ]]; then
+  change_count=$(git diff --merge-base $base_ref --name-only |awk -F"/" '{ print $1}' |uniq |wc -l)
```
Comment (joshbrand) [inline .github/scripts/verify_changes.sh:11]: formatting nit, would make it a little clearer to follow with a space after the pipes as well!

File: .github/scripts/verify_changes.sh
Comment (joshbrand) [inline .github/scripts/verify_changes.sh:1]: GHA dumb question: where/how does this get called? I'm not seeing it or not awake enough yet

Comment (jaymalasinha) [reply .github/scripts/verify_changes.sh:1]: https://github.com/hashicorp/vault/pull/20036/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R27

File: .github/scripts/verify_changes.sh (hunk 1 above)
Comment (jaymalasinha) [reply .github/scripts/verify_changes.sh]: Fixed 👍

**Review by joshbrand — APPROVED** [review]
😸

**Pull request merged**